### PR TITLE
Feat: Update K8s core

### PR DESCRIPTION
**Pull request by dnitsch**
#### 📲 What

Updated TF container to 0.12.24
updated TF azurerm provider to 2.x
updated AKS to use RBAC
added AppGateway integration

TODO: add SSL options and biding to frontend app gateway

#### 🤔 Why
		
Why it's needed, background context.
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 Evidence
		
Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR
		 
#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: libs/images/tf-deploy/README.md [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
+# Terraform deployment container
 
-provides a Kubectl cli  and Kustomize
+Provides a Terraform CLI
 
 versions:
-  - kubectl: v1.15.1
-  - kustomize v3.5.4
-
+  - terraform: 0.12.24
```
Comment (estherthetesteramido) [inline libs/images/tf-deploy/README.md]: nice

File: libs/orchestration/terraform-azurerm-amido-aks/aks.tf [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      key_data = chomp(tls_private_key.ssh_key[0].public_key_openssh)
+    }
+  }
+
+  default_node_pool {
+    # TODO: variablise below:
+    availability_zones = ["1", "2", "3"] # var.aks_azs
+    # node_taints           = [] -> null
+    # max_pods          = var.max_pods != 0 ? var.max_pods : 1
+    # enable_node_public_ip = false
```
Comment (estherthetesteramido) [inline libs/orchestration/terraform-azurerm-amido-aks/aks.tf]: these to be verbalised? or removed?

File: libs/orchestration/terraform-azurerm-amido-aks/aks.tf [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    type                = var.nodepool_type # "VirtualMachineScaleSets" # default
+    enable_auto_scaling = var.enable_auto_scaling
+    max_count           = var.max_nodes
+    min_count           = var.min_nodes
+    name                = "default"
+    os_disk_size_gb     = var.os_disk_size
+    vm_size             = var.vm_size
+    node_count          = var.min_nodes
+    # vnet_subnet_id      = var.vnet_subnet_id
```
Comment (estherthetesteramido) [inline libs/orchestration/terraform-azurerm-amido-aks/aks.tf]: same q as above

File: libs/orchestration/terraform-azurerm-amido-aks/examples/entire-infra/constraints.tf [hunk 4]
```diff
@@ -1,7 +1,7 @@
 terraform {
   required_version = ">= 0.12"
   required_providers {
-    azurerm = "~> 1.44.0"
+    azurerm = "~> 2"
```
Comment (estherthetesteramido) [inline libs/orchestration/terraform-azurerm-amido-aks/examples/entire-infra/constraints.tf]: No point version?

File: packages/template-cli/templates/deploy/azure/ssr/main.tf [hunk 5]
```diff
@@ -20,7 +20,7 @@ variable "vnet_cidr" {
 }
 
 module "sample_aks_bootstrap" {
-  source                  = "git::https://github.com/amido/stacks-webapp-template//libs/orchestration/terraform-azurerm-amido-aks?ref=feat/1357"
+  source                  = "git::https://github.com/amido/stacks-webapp-template//libs/orchestration/terraform-azurerm-amido-aks?ref=feat/1435"
```
Comment (estherthetesteramido) [inline packages/template-cli/templates/deploy/azure/ssr/main.tf]: Should this reference master?

Comment (dnitsch) [reply packages/template-cli/templates/deploy/azure/ssr/main.tf]: it should once we are ready to release it - it shold actually reference a release tag :) 
but for us for testing we can reference self  branch in monorepo

File: libs/orchestration/terraform-azurerm-amido-aks/examples/entire-infra/constraints.tf (hunk 4 above)
Comment (dnitsch) [reply libs/orchestration/terraform-azurerm-amido-aks/examples/entire-infra/constraints.tf]: let's stick to latest 2.x version for now - it just shouldn't be 1.x

File: libs/orchestration/terraform-azurerm-amido-aks/README.md [hunk 6]
```diff
@@ -69,64 +67,68 @@ terraform init -backend-config=./backend.local.tfvars
...
+| vnet\_name | VNET name if create\_aks\_vnet is false | `string` | `"changeme"` | no |
 
 ## Outputs
 
-No output.
+| Name | Description |
+|------|-------------|
+| aks\_default\_user\_identity\_client\_id | n/a |
+| aks\_default\_user\_identity\_id | n/a |
+| aks\_default\_user\_identity\_name | ######################################## ############ Identity ################## ## used for AAD Pod identity binding ### ######################################## |
```
Comment (estherthetesteramido) [inline libs/orchestration/terraform-azurerm-amido-aks/README.md:122]: To change someday....

**Pull request merged**